### ISSUE TITLE
[5.4] Storage fake doesn't work well with response()->download()

### ISSUE THREAD

**Issue by santigarcor** [laravel/framework#20379]
- Laravel Version: 5.4.30
- PHP Version: 7.1

### Description:
If you are testing file generation and download using the `Storage::fake`, when you pass the file url to the download response, it doesn't find the file because the url method returns '/storage/filename'.

It should have an option to create the fake so it sets the proper `url` to the fake config.

### Steps To Reproduce:
In your test:
```php
Storage::fake('thedisk');
```

In your controller:
```php
// Some file creation logic
// ...

return response()->download(
    Storage::disk('thedisk')->url($createdFileName),
    'somefile.xls',
    ['Content-Type' => 'application/vnd.ms-excel']
);
```

And it will throw:
`Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException: The file "/storage/thecreatedfile.xls" does not exist`

**Comment by themsaid** [comment]
Please share how you create the file, what's in your `$createdFileName` variable. 

Sharing incomplete steps of replication causes lots of confusion.

**Comment by santigarcor** [comment]
I'm using the [Maatwebsite Excel](https://github.com/Maatwebsite/Laravel-Excel) package:
```php
Excel::create('template', function ($excel) {
        $excel->sheet('something', function ($sheet) {
            $sheet->fromArray(['a', 'b']);
        });
    })->store(
        'xls',
        Storage::disk('thedisk')
        ->getAdapter()
        ->applyPathPrefix($model->id)
   );
```
During the test that code stores the file inside `framework/testing/disks/datasets/1/template.xls` where `1` is the model id, and then for the `$createdFileName` just:

```php
$createdFileName = "{$model->id}/template.xls";
```

It doesn't work when calling the url method to that `$createdFileName`, so the workaround i found to get the test to green was:

```php
$createdFileName = Storage::disk('thedisk')
        ->getAdapter()
        ->applyPathPrefix("{$model->id}/template.xls");

return response()->download(
    $createdFileName,
    'somefile.xls',
    ['Content-Type' => 'application/vnd.ms-excel']
);

```

**Comment by themsaid** [comment]
Hmmm, I don't think `url()` should be used here, I've proposed a new method: https://github.com/laravel/framework/pull/20395

**Comment by santigarcor** [comment]
Yes, that `path` or `realPath` method is a better than the `url()`.

**Comment by themsaid** [comment]
That was merged in 5.4, will be available in next release :)

**Comment by santigarcor** [comment]
Thanks!
